lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -962,3 +765,44 @@ evaluateTransactionBalance era pp utxo = withConstraints era $
         -- Pass this parameter in as a function instead of hard-coding the
         -- value here:
         const True
+
+
+--------------------------------------------------------------------------------
+-- Constraints
+--------------------------------------------------------------------------------
+
+-- | Convenient constraints. Constraints may be dropped as we move to new eras.
+--
+-- Adding too many constraints shouldn't be a concern as the point of 'RecentEra'
+-- is to work with a small closed set of eras, anyway.
+type RecentEraLedgerConstraints era =
+    ( Core.Era era
+    , Core.Script era ~ AlonzoScript era
+    , CLI era
+    , Crypto era ~ StandardCrypto
+    , Core.Tx era ~ Babbage.AlonzoTx era
+    , Core.Value era ~ MaryValue StandardCrypto
+    , Babbage.ShelleyEraTxBody era
+    , HasField' "_collateralPercentage" (Core.PParams era) Natural
+    , HasField' "_maxCollateralInputs" (Core.PParams era) Natural
+    , HasField' "_minfeeA" (Core.PParams era) Natural
+    , HasField' "_prices" (Core.PParams era) ExUnitPrices
+    , HasField' "_maxTxExUnits" (Core.PParams era) ExUnits
+    , HasField' "_keyDeposit" (Core.PParams era) Coin
+    )
+
+-- | Bring useful constraints into scope from a value-level
+-- 'RecentEra'.
+--
+-- TODO [ADP-2354] Make 'RecentEraLedgerConstraints' superclass of
+-- 'IsRecentEra'. Currently this would cause weird type-inference errors in the
```

Comment:
> 'RecentEraLedgerConstraints' superclass

This is the downside of splitting it out. The only way to have `RecentEraLedgerConstraints` as a superclass would be by also defining `RecentEraLedgerConstraints` in the `Era` module, which would require importing all of these constraints and making the separation between `Write.Era` and `Write.Tx` fuzzier.